engines/scumm/he/moonbase/map_spiff.cpp

```diff
@@ -0,0 +1,623 @@
+	// --------------------------------------------------------------
+	int x;
+	int y;
+	int neighbors[MAXELEVVAL];
+	int a;
+	int b;
+	int tempElevation;
+	int nextElevation;
+	int special;
+
+	mapCornerMaxG = totalMapSizeG / 2;
+	mapMiddleMaxG = mapCornerMaxG - 1;
+
+	for (y = 0; y <= mapCornerMaxG; ++y) {
+		// initialise map to UNASSIGNED tiles
+		for (x = 0; x <= mapCornerMaxG; ++x) {
+			mapCorner[x][y] = UNASSIGNED;
+			mapMiddle[x][y] = UNASSIGNED;
+		}
+	}
+	if (advancedMirrorOK_G)
+		mirrorTypeG = pickFrom4(NORMALMIRROR, 1, XOFFSETMIRROR, 2, YOFFSETMIRROR, 2, MAXDISTMIRROR, 4);
+	else
+		mirrorTypeG = NORMALMIRROR;
+	getSpecials(); // get start and pools
+
+	// --------------------------------------------------------------
+	//  loop through each square
+	// --------------------------------------------------------------
+	mapCorner[0][0] = pickFrom3(LOW, 1, MEDIUM, (terrainSeedFlagG < 9), HIGH, (terrainSeedFlagG < 8)); // seed
+	// fill in the rest of the random map
+	for (y = 0; y <= mapCornerMaxG; ++y) {
+		for (x = 0; x <= mapCornerMaxG; ++x) {
+			special = mapMiddle[x][y]; // water wouldn't have been assigned yet, so must be pool, start, or UNASSIGNED
+
+			// --------------------------------------------------------------
+			//  check neighbors
+			// --------------------------------------------------------------
+			if ((mapCorner[x][y] != UNASSIGNED) && (mapCorner[x][y] != LOW_OR_WATER))
+				nextElevation = mapCorner[x][y]; // already defined because of a special or (0,0), so no change
+			else {
+				neighbors[HIGH] = 0;
+				neighbors[MEDIUM] = 0;
+				neighbors[LOW] = 0;
+				neighbors[WATER] = 0;
+
+				if (x > 0) {
+					a = mapCorner[x - 1][y];
+					if ((y > 1) && (mapMiddle[x - 1][y - 2] == WATER))
+						++neighbors[WATER];
+					if (y > 0)
+						neighbors[mapCorner[x - 1][y - 1]] += 3;
+				} else {
+					a = mapCorner[x][y - 1];
+				}
+
+				neighbors[a] += 3;
+				if (y > 0) {
+					b = mapCorner[x][y - 1];
+					neighbors[b] += 3;
+					if (x < mapCornerMaxG) {
+						++neighbors[mapCorner[x + 1][y - 1]]; // so this value can be ignored when choosing water
+						if ((special != UNASSIGNED) && (x < mapCornerMaxG - 1))
+							++neighbors[mapCorner[x + 2][y - 1]];
+					}
+					if ((x > 1) && (mapMiddle[x - 2][y - 1] == WATER))
+						++neighbors[WATER];
+				} else {
+					b = mapCorner[x - 1][y]; // for probability equations for edges
+				}
+
+				// --------------------------------------------------------------
+				//  pick new elevation
+				// --------------------------------------------------------------
+				// neighbors                  possible new elevation
+				// HIGH or HIGH with MEDIUM   HIGH or MEDIUM
+				// MEDIUM only                HIGH, MEDIUM or LOW
+				// LOW or WATER only          MEDIUM, LOW or WATER
+				// MEDIUM with LOW or WATER   MEDIUM or LOW, possible WATER if no MEDIUM left, down, or down-left
+				// HIGH with LOW or WATER     MEDIUM
+
+				static int highAmt = 105;
```

Comment:
why are these declared static?

Reply:
These were [originally `#define`s on Moonbase Console](https://github.com/michaelbarlow7/moonbase-console/blob/master/MoonbaseConsole/SpiffGen.cpp#L430-L433).  You want them non-static or reverted as `#define`s?